### PR TITLE
server: Add player.spawn(model, pos) overload

### PR DESCRIPTION
**Pull request by xxshady**
Adds `player.spawn(model: string | number, pos: alt.IVector3)`

### CODE REVIEW

File: server/src/bindings/Player.cpp [hunk 1]
```diff
@@ -26,16 +26,43 @@ static void Spawn(const v8::FunctionCallbackInfo<v8::Value>& info)
...
-        V8_ARG_TO_VECTOR3(1, pos);
-        if(info.Length() == 2)
+        alt::Vector3f pos;
+
+        // (model: number | string, pos: IVector3) overload
+        if(info[0]->IsNumber() || info[0]->IsString())
         {
-            V8_ARG_TO_UINT(2, spawnDelay);
-            delay = spawnDelay;
+            V8_CHECK_ARGS_LEN(2);
```
Comment (LeonMrBonnie) [inline server/src/bindings/Player.cpp:37]: there are macros for args check with 2 options or min max, use those at the top of the function

File: server/src/bindings/Player.cpp [hunk 2]
```diff
@@ -26,16 +26,43 @@ static void Spawn(const v8::FunctionCallbackInfo<v8::Value>& info)
...
+
+            if(info[0]->IsNumber())
+            {
+                V8_TO_INTEGER(info[0], model);
```
Comment (LeonMrBonnie) [inline server/src/bindings/Player.cpp:41]: use the macro for uint, because model is unsigned

File: server/src/bindings/Player.cpp [hunk 3]
```diff
@@ -26,16 +26,43 @@ static void Spawn(const v8::FunctionCallbackInfo<v8::Value>& info)
...
+                _this->SetModel(model);
+            }
+            else
+            {
+                V8_TO_STRING(info[0], model);
+                _this->SetModel(alt::ICore::Instance().Hash(model));
+            }
+
+            V8_CHECK(V8Helpers::SafeToVector3(info[1], ctx, pos), "Failed to convert argument 2 to vector3");
```
Comment (LeonMrBonnie) [inline server/src/bindings/Player.cpp:50]: there is a macro for safely getting a vector3 arg

File: server/src/bindings/Player.cpp (hunk 1 above)
Comment (xxshady) [reply server/src/bindings/Player.cpp:37]: What do you mean? There is already macro for 1-4 args
https://github.com/altmp/altv-js-module/blob/65ca93fa7440a407cd034c28948559bbe7c855ee/server/src/bindings/Player.cpp#L23

Comment (xxshady) [reply server/src/bindings/Player.cpp:37]: But I need exactly 2 args here

Comment (LeonMrBonnie) [reply server/src/bindings/Player.cpp:37]: then you just ignore all other args, dont add two checks for arg length

File: server/src/bindings/Player.cpp (hunk 2 above)
Comment (xxshady) [reply server/src/bindings/Player.cpp]: Done

File: server/src/bindings/Player.cpp (hunk 1 above)
Comment (xxshady) [reply server/src/bindings/Player.cpp]: Done

File: server/src/bindings/Player.cpp (hunk 3 above)
Comment (xxshady) [reply server/src/bindings/Player.cpp]: Done

**Pull request merged**